c/src/ssl/schannel.cpp

```diff
@@ -1765,6 +1765,12 @@ static ssize_t process_output_ssl( pn_transport_t *transport, unsigned int layer
   if (!ssl) return PN_EOS;
   ssl_log( transport, PN_LEVEL_TRACE, "process_output_ssl( max_len=%d )",max_len );
 
+  // This is to match the behaviour of pn_output_write_amqp defined in transport.c.
+  // Without this, the idle_timeout connection option does not work in case of an SSL connection.
+  if (!pn_buffer_size(transport->output_buffer) && transport->close_sent) {
+    return PN_EOS;
+  }
+
```

Comment:
I'm somewhat confused about this change - the equivalent openssl code does not do anything like this but as far as I know works correctly in the face of idle timeout. This makes me think that this is not the correct fix to the problem.
Especially as the check should be redundent in that the amqp layer processing which does have this check will still be running and should pick up on the issue. Clearly there is an issue somewhere that is specific to the schannel implementation, but this seems like a 'voodoo' fix.
@cliffjansen @kgiusti do either of you have any thoughts?